### ISSUE TITLE
C++: In brace initializer, first arithmetic operand to * mistaken for a type

### ISSUE THREAD

**Issue by kintel** [uncrustify/uncrustify#1466]
Similar to #1464, but looks like a separate issue:

In some cases, the first arithmetic operand to `*` is mistaken for a type and the spacing gets messed up, when the code is inside a brace initializer:

```
class ClassA
{
ClassA() {
	std::pair<float, float> a = {this->r * cos(this->a), this->r * sin(this->a)};
}
float r{1.0f};
float a{1.57f};
};
```

Results in (notice the space after the first `*` gets removed):
```
std::pair<float, float> a = {this->r *cos(this->a), this->r * sin(this->a)};
```

uncrustify.cfg:
```
sp_arith=force
sp_after_ptr_star=remove
```

[debug.txt](https://github.com/uncrustify/uncrustify/files/1518799/debug.txt)

Works as expected in 0.65, but fails as described above in 0.66 and newer, including latest master (`Uncrustify-0.66.1-32-39e43b55`) and latest master with suggested fix for #1464 applied.

**Comment by CDanU** [comment]
This one is pretty much the same as the previous issue, try:
```diff
diff --git a/src/combine.cpp b/src/combine.cpp
index 04c8f6b8..0307a408 100644
--- a/src/combine.cpp
+++ b/src/combine.cpp
@@ -4199,6 +4199,10 @@ static void mark_function(chunk_t *pc)
             }
          }
 
+         if(isa_def && prev->type == CT_MEMBER)
+         {
+             isa_def = false;
+         }
 
          // Skip the word/type before the '.' or '::'
          if (prev->type == CT_DC_MEMBER || prev->type == CT_MEMBER)
```

**Comment by kintel** [comment]
@CDanU Yeps, that fixes it everywhere!

**Comment by kintel** [comment]
Fixed by 91a46463a96ad455733f2e06db6a50f7bd93da0f

@CDanU Any chance for a new release soon?

**Comment by CDanU** [comment]
@kintel Unlikely, the last time we released was in November.  
Ultimately that's up to gmaurel, but so far we have been maintaining more or less a six month release schedule.
